src/openrct2/world/map.c

```diff
@@ -3168,11 +3168,16 @@ void game_command_place_large_scenery(sint32* eax, sint32* ebx, sint32* ecx, sin
 
 sint32 map_get_station(rct_map_element *mapElement)
 {
-    sint32 station = (mapElement->properties.track.sequence & 0x70) >> 4;
-
     // prevent out of bounds access
-    clamp(0, station, RCT12_MAX_STATIONS_PER_RIDE - 1);
-    return station;
+    sint32 station = (mapElement->properties.track.sequence & SEQUENCE_GET_STATION) >> 4;
+    sint32 clamped_station = clamp(0, station, RCT12_MAX_STATIONS_PER_RIDE - 1);
+    if (clamped_station != station)
+    {
+        log_error("Invalid savestate corrected: current_station-value clamped to %d (was %d)", clamped_station, station);
+        mapElement->properties.track.sequence &= 0x8F;
+        mapElement->properties.track.sequence |= clamped_station << 4;
+    }
+    return clamped_station;
```

Comment:
there was no assert here before but it might be added for completeness.
I also tried looking into the importer, however the mapElements are read as one big chunk that would need to be checked somehow.